dotnet/src/Planners/Planners.Handlebars/Handlebars/HandlebarsPlanner.cs

```diff
@@ -153,7 +161,26 @@ private async Task<HandlebarsPlan> CreatePlanCoreAsync(Kernel kernel, string goa
         HashSet<HandlebarsParameterTypeMetadata> complexParameterTypes,
         Dictionary<string, string> complexParameterSchemas)
     {
-        // TODO (@teresaqhoang): Handle case when schema and ParameterType can exist i.e., when ParameterType = RestApiResponse
+        if (parameter.Schema is not null)
+        {
+            // Class types will have a defined schema, but we want to handle those as built-in complex types below
+            if (parameter.ParameterType is not null && parameter.ParameterType!.IsClass)
+            {
+                parameter = new(parameter) { Schema = null };
+            }
+            else
+            {
+                // Parse the schema to extract any primitive types and set in ParameterType property instead
+                var parsedParameter = parameter.ParseJsonSchema();
+                if (parsedParameter.Schema is not null)
+                {
+                    complexParameterSchemas[parameter.GetSchemaTypeName()] = parameter.Schema.RootElement.ToJsonString();
+                }
+
+                return parsedParameter;
```

Comment:
Single return would be better

Reply:
It's better to return here to avoid further unnecessary processing. Before the second conditional was encompassed in an `else if` so single return worked, but there can be scenarios with Schema and ParameterType can co-exist and we explicitly handle one scenario above, but for all other scenarios, we only want to prioritize and process the schema only and skip the ParameterType